src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ConnectionPool/IDbConnectionPool.cs

```diff
@@ -0,0 +1,150 @@
+        /// </summary>
+        int Count { get; }
+
+        /// <summary>
+        /// Gets the factory used to create database connections.
+        /// </summary>
+        DbConnectionFactory ConnectionFactory { get; }
+
+        /// <summary>
+        /// Indicates whether an error has occurred in the pool.
+        /// Primarily used to support the pool blocking period feature.
+        /// </summary>
+        bool ErrorOccurred { get; }
+
+        /// <summary>
+        /// Gets the duration of time to wait before reassigning a connection to a different server in a load-balanced
+        /// environment.
+        /// </summary>
+        TimeSpan LoadBalanceTimeout { get; }
+
+        /// <summary>
+        /// Gets the identity used by the connection pool when establishing connections.
+        /// </summary>
+        DbConnectionPoolIdentity Identity { get; }
+
+        /// <summary>
+        /// Indicates whether the connection pool is currently running.
+        /// </summary>
+        bool IsRunning { get; }
+
+        /// <summary>
+        /// Gets a reference to the connection pool group that this pool belongs to.
+        /// </summary>
+        DbConnectionPoolGroup PoolGroup { get; }
+
+        /// <summary>
+        /// Gets the options for the connection pool group.
+        /// </summary>
+        DbConnectionPoolGroupOptions PoolGroupOptions { get; }
+
+        /// <summary>
+        /// Gets the provider information for the connection pool.
+        /// </summary>
+        DbConnectionPoolProviderInfo ProviderInfo { get; }
+
+        /// <summary>
+        /// Gets the authentication contexts cached by the pool.
+        /// </summary>
+        ConcurrentDictionary<DbConnectionPoolAuthenticationContextKey, DbConnectionPoolAuthenticationContext> AuthenticationContexts { get; }
+
+        /// <summary>
+        /// Indicates whether the connection pool is using load balancing.
+        /// </summary>
+        bool UseLoadBalancing { get; }
+        #endregion
+
+        #region Methods
+        /// <summary>
+        /// Clears the connection pool, releasing all connections and resetting the state.
+        /// </summary>
+        void Clear();
+
+        /// <summary>
+        /// Attempts to get a connection from the pool.
+        /// </summary>
+        /// <param name="owningObject">The SqlConnection that will own this internal connection.</param>
+        /// <param name="taskCompletionSource">Used when calling this method in an async context. 
+        /// The internal connection will be set on completion source rather than passed out via the out parameter.</param>
+        /// <param name="userOptions">The user options to use if a new connection must be opened.</param>
+        /// <param name="connection">The retrieved connection will be passed out via this parameter.</param>
+        /// <returns>Returns true if a connection was set in the out parameter, otherwise returns false.</returns>
```

Comment:
Super Nit: It always seems redundant to me have a <returns> tag and then starting the description as "Returns xyz"...

Same could be said about starting descriptions of parameters with "the". But that is just me :)